lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,399 @@
+
+        filename = self.output_irf_file.name
+        if filename.split(".")[1] != "fits":
+            self.log.warning(
+                f"{filename} is not a correct "
+                "compressed FITS file name. It will be corrected."
+            )
+            filename = filename.split(".")[0] + ".fits.gz"
+            self.output_irf_file = self.output_irf_file.parent / filename
+
+        if self.input_proton_dl2 and self.input_electron_dl2 is not None:
+            self.only_gamma_irf = False
+        else:
+            self.only_gamma_irf = True
+
+        self.data_sel = DataSelection(parent=self)
+        self.data_bin = DataBinning(parent=self)
+
+        self.mc_particle = {
+            "gamma": {
+                "file": str(self.input_gamma_dl2),
+                "target_spectrum": CRAB_HEGRA,
+            },
+        }
+        Provenance().add_input_file(self.input_gamma_dl2)
+
+        # Read and update MC information
+        if not self.only_gamma_irf:
+            self.mc_particle["proton"] = {
+                "file": str(self.input_proton_dl2),
+                "target_spectrum": IRFDOC_PROTON_SPECTRUM,
+            }
+
+            self.mc_particle["electron"] = {
+                "file": str(self.input_electron_dl2),
+                "target_spectrum": IRFDOC_ELECTRON_SPECTRUM,
+            }
+
+            Provenance().add_input_file(self.input_proton_dl2)
+            Provenance().add_input_file(self.input_electron_dl2)
+
+        self.provenance_log = self.output_irf_file.parent / (
+            self.name + ".provenance.log"
+        )
+
+    def start(self):
+
+        for particle_type, p in self.mc_particle.items():
+            self.log.info(f"Simulated {particle_type.title()} Events:")
+            p["events"], p["simulation_info"] = read_mc_dl2_to_QTable(p["file"])
+
+            if p["simulation_info"].viewcone.value == 0.0:
+                p["mc_type"] = "point_like"
+            else:
+                p["mc_type"] = "diffuse"
+                # For diffuse gamma using Proton Spectra for calculating event weights
+                if particle_type == "gamma":
+                    if not self.point_like:
+                        p["target_spectrum"] = IRFDOC_PROTON_SPECTRUM
+                        self.log.debug(
+                            "Proton spectrum used as target spectrum"
+                            " for MC diffuse gamma"
+                        )
+                    else:
+                        raise ToolConfigurationError(
+                            "Diffuse MC gamma cannot be used for generating "
+                            "point_like IRFs. Use appropriate MC and IRF type."
+                        )
+
+            self.log.debug(f"Simulated {p['mc_type']} {particle_type.title()} Events:")
+
+            p["simulated_spectrum"] = PowerLaw.from_simulation(
+                p["simulation_info"], 50 * u.hour
+            )
+            p["events"]["weight"] = calculate_event_weights(
```

Comment:
You only need this for the background, not for gamma-rays. Setting  wrong spectra for gammas just so you can compute weights does not make sense.